worksheet.rb

```diff
@@ -1,34 +1,121 @@
-
-# Use an iteration blocks to print the following answers:
-# - the number of rides each driver has given
-# - the total amount of money each driver has made
-# - the average rating for each driver
-# - Which driver made the most money?
-# - Which driver has the highest average rating?
\ No newline at end of file
+drivers = {"DR0001": {
+                        "3rd Feb 2016": [
+                            {cost: 10, rider_id: "RD0003", rating: 3},
+                            {cost: 30, rider_id: "RD0015", rating: 4}
+                        ],
+                        "5th Feb 2016": [
+                             {cost: 45, rider_id: "RD0003", rating: 2}
+                        ]
+                      },
+           "DR0002": {
+                        "3rd Feb 2016": [
+                            {cost: 25, rider_id: "RD0073", rating: 5}
+                        ],
+                        "4th Feb 2016": [
+                            {cost: 15, rider_id: "RD0013", rating: 1}
+                        ],
+                        "5th Feb 2016": [
+                            {cost: 35, rider_id: "RD0066", rating: 3}
+                        ]
+                      },
+           "DR0003": {
+                        "4th Feb 2016": [
+                            {cost: 5, rider_id: "RD0066", rating: 5},
+                        ],
+                        "5th Feb 2016": [
+                            {cost: 50, rider_id: "RD0003", rating: 2}
+                        ]
+                      },
+           "DR0004": {
+                        "3rd Feb 2016": [
+                            {cost: 5, rider_id: "RD0022", rating: 5}
+                        ],
+                        "4th Feb 2016": [
+                            {cost: 10, rider_id: "RD0022", rating: 4}
+                        ],
+                        "5th Feb 2016": [
+                            {cost: 20, rider_id: "RD0073", rating: 5}
+                        ]
+                      }
+          }
+
+#Q4
+def richest(drivers)
+  earnings = []
+  drivers.each do |driver, data|
+    total = []
+    data.each do |day, rides|
+      rides.each do |ride|
+        total.push ride[:cost]
+      end
+    end
+    earnings.push total.sum
+    if total.sum == earnings.max
+      puts "\nDriver #{driver} made the most money for the period."
+    end
+  end
+end
+
+#Q5
+def best_rating(drivers)
+  highest_average = []
+  which_driver = []
+  drivers.each do |driver, data|
+    av_rates = []
+    data.each do |day, rides|
+      rides.each do |ride|
+        av_rates.push ride[:rating]
+      end
+    end
+    which_driver.push driver
+    highest_average.push av_rates.sum.to_f / av_rates.length
+  end
+  if highest_average.include?(highest_average.max)
+    who = highest_average.index(highest_average.max)
+    puts "Driver #{which_driver[who]} has the highest average rating."
+  end
+end
+
+#All questions answered
+def ride_share(drivers)
```

Comment:
Note that you separated out questions 4 and 5 into their own methods, and do the work for questions 1-3 in the method `ride_share` where you also call the methods to answer questions 4 and 5. Consider creating one method for questions 1-3, such as `calculate_individual_driver_stats` and then another method that's focused on user output that calls `calculate_individual_driver_states`, `best_rating`, and `richest`.